testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js

```diff
@@ -2180,6 +2180,35 @@ QUnit.module('keyboard navigation', {
         fx.off = false;
     }
 }, () => {
+    QUnit.test('navigation keys should focus items', function(assert) {
+        const $listItems = getListItems(this.instance);
+
+        assertFocusedItem(assert, this.keyboard, 'down', $listItems, 1);
+        assertFocusedItem(assert, this.keyboard, 'down', $listItems, 2);
+        assertFocusedItem(assert, this.keyboard, 'up', $listItems, 1);
+        assertFocusedItem(assert, this.keyboard, 'up', $listItems, 0);
+        assertFocusedItem(assert, this.keyboard, 'pagedown', $listItems, 2);
+        assertFocusedItem(assert, this.keyboard, 'pageup', $listItems, 0);
+    });
+
+    function assertFocusedItem(assert, keyboard, key, listItems, expectedFocusedItemIndex) {
```

Comment:
1. We can simplify signature if this function If we move it to "beforeEach" and save as an instance's method. Then we don't need to pass `keyboard` and `listItems` props
2. This function's name is not complete. We don't only assert focused item, we also press a key here. So it should be named as `pressKeyAndAssertFocusedItem`. Here u can already see that there is an other issue - this function is not single responsible. It's better to make an action and an assert separately.

So, i suggest u get rid of this function and write this test imperatively. Yeap, there would be some copypaste, but the tests would be easier to read (especially, after splitting).

```javascript
QUnit.test('down/up key press should move item focus', function(assert) {
       const $listItems = getListItems(this.instance);
       const $firstItem = $listItems.eq(0);
       const $secondItem = $listItems.eq(1);
       
       this.keyboard.press('down');
       assert.strictEqual($secondItem.hasClass(FOCUSED_CLASS), 'second item is focused');

       this.keyboard.press('up');
       assert.strictEqual($firstItem.hasClass(FOCUSED_CLASS), 'first item is focused');
});
```